azuredevops/internal/acceptancetests/resource_serviceendpoint_github_test.go

```diff
@@ -53,3 +173,63 @@ func TestAccServiceEndpointGitHub_CreateAndUpdate(t *testing.T) {
 		},
 	})
 }
+
+func personTokenConfigBasic(projectName string, serviceEndpointName string) string {
+	projectResource := testutils.HclProjectResource(projectName)
+
+	serviceEndpointResource := fmt.Sprintf(`
+resource "azuredevops_serviceendpoint_github" "serviceendpoint" {
+	project_id             = azuredevops_project.project.id
+	service_endpoint_name  = "%[1]s"
+	auth_personal {
+		personal_access_token= "xxxxxxxxxxxxxxxxxxxxxxxxxx"
+	}
+}`, serviceEndpointName)
+
+	return fmt.Sprintf("%s\n%s", projectResource, serviceEndpointResource)
+}
+
+func personTokenConfigUpdate(projectName string, serviceEndpointName string, description string) string {
+	projectResource := testutils.HclProjectResource(projectName)
+
+	serviceEndpointResource := fmt.Sprintf(`
+resource "azuredevops_serviceendpoint_github" "serviceendpoint" {
+	project_id             = azuredevops_project.project.id
+	service_endpoint_name  = "%[1]s"
+	auth_personal {
+		personal_access_token= "xxxxxxxxxxxxxxx"
+	}
+	description = "%[2]s"
+}`, serviceEndpointName, description)
+
+	return fmt.Sprintf("%s\n%s", projectResource, serviceEndpointResource)
+}
+
+func oauthConfigBasic(projectName string, serviceEndpointName string) string {
+	projectResource := testutils.HclProjectResource(projectName)
+	serviceEndpointResource := fmt.Sprintf(`
+resource "azuredevops_serviceendpoint_github" "serviceendpoint" {
+	project_id             = azuredevops_project.project.id
+	service_endpoint_name  = "%[1]s"
+	auth_oauth {
+		oauth_configuration_id= "xxxx-xxxx-xxxx-xxxx-xxxx"
+	}
+}`, serviceEndpointName)
+
+	return fmt.Sprintf("%s\n%s", projectResource, serviceEndpointResource)
+}
+
+func oauthConfigUpdate(projectName string, serviceEndpointName string, description string) string {
```

Comment:
Do we not want to put them into the hcl.go anymore?

Reply:
I plan to refactor the code
1. The `HCL` test configure is used only in`resource_serviceendpoint_github_test.go`,  no need to make this test data public
2. All test `HCL` in one file, soon we will get a monster file, keep the test data where is belongs will be more clear.

Reply:
I agree but you still need some place where you put shared HCL definitions. If you add the HCL functions in the `_test.go` files you'll clutter up them as well. Many `_test.go` files are already huge!